### ISSUE TITLE
[Feature Request]:  Adapt the Docker Deployment to be more cloud friendly 

### ISSUE THREAD

**Issue by m4us1ne** [grokability/snipe-it#13214]
### Is your feature request related to a problem? Please describe.

snipe-it in docker is works a bit like a vm in a docker container. it installs lots of packages (some of them are not needed) to run the app itself.  the entrypoint is not the app itself, but supervisord. this puts another layer between the app itself and k8s/docker.

### Describe the solution you'd like

Ideally we build no "all in one" container images, but seperate images for the webserver, fpm, cli and schedule task via kubernetes cron or a scheduling container.
ideally this deployment  follows the 8 container principles https://kubernetes.io/blog/2018/03/principles-of-container-app-design/
maybe this is worth a read/skim https://chris-vermeulen.com/laravel-in-kubernetes-part-1/

### Additional context

the main benefit of this change would be scalability and resilience. kubernetes or another container hypervisor (docker, ....) can identify malfunctioning parts of the app. more requests would just mean another fpm container handling requests.

i want to contribute that to snipe-it, but i am unsure if this would be a welcomed change. if you could give me feedback i would start working on this.

**Comment by snipe** [comment]
@uberbrady any thoughts?

**Comment by uberbrady** [comment]
I'm on the fence about this. On the one hand, some people (like @m4us1ne, but also @inietov !) do really want stuff like this. But I would only want it as another optional Dockerfile or set of Dockerfiles (similar to how we do for Alpine). Or maybe it'd be a docker-compose file? Maybe both? But I'm really nervous about whether or not we would be able to *support* it - we definitely won't be able to support it for our pro-support customers, and I'm not comfortable with us having a platform that we can't offer as pro-support. I mean, I guess we could support it with Enterprise Support? 

But regardless, we'd have to probably add a few more build steps to our build automation that we do so that this would get auto-built and pushed up to Dockerhub, too. So that's more work on our end :/

From a technological perspective, having a separate nginx container versus an FPM container feels like you'd have two copies of the codebase? Well, you'd also need another one for the cron container, and for the queue-runner container. At least, for a production environment, you would. During development you can certainly stick with the volume-mounts, but it feels like that's not appropriate for a production environment, right? Or maybe that's the point that I'm stuck on, incorrectly.

Additionally, I think the codebase would need some tweaking to work like this - logging to stdout, for example - and we'd probably have to rip out a lot of the olden-skoole `--link` options (which are deprecated in Docker, but I suspect doesn't even *work* in Kubernetes).

So, to summarize - I'm very much on the fence :/

**Comment by StarlessNights** [comment]
Maybe a reasonable first step in this direction would be to direct apache logs to stdout/stderr?

Currently they just end up buried in the standard /var/log/apache2/* files. (in the ubuntu container at least)

Supervisor seems to configured for this already: https://github.com/snipe/snipe-it/blob/master/docker/supervisord.conf

So configuring apache should be sufficient, but there also seems to be a laravel specific way in the blog series from the original post: https://chris-vermeulen.com/laravel-in-kubernetes-part-2/#logging-update

I don't understand enough laravel to asses the possible merits of going this way. Thoughts?

**Comment by uberbrady** [comment]
That is *very* reasonable. I've been working on the Dockerfiles for a few weeks now, hoping to modernize, simplify, shrink, and generally improve them. I'll make sure that logging to stdout is in that list.

**Comment by floptical** [comment]
@uberbrady was this ever worked on? The latest version of the docker image (8.6.1) does not appear to send errors to stdout/stderr (as seen with `docker logs <container>`) when the LOG_CHANNEL env var is set to "stderr". It does appear to prevent the "storage/logs/laravel.log" file from being made so it's doing something.
